### PR TITLE
feature-request: Added API Gateway WAF Enabled plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/apigateway/apigatewayWafEnabled.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'API Gateway WAF Enabled',
+    category: 'API Gateway',
+    description: 'Ensures that API Gateway APIs are associated with Web Application Firewall.',
```
Comment (dipsubha06) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:7]:
```suggestion
    description: 'Ensures that API Gateway APIs are associated with a Web Application Firewall.',
```

File: plugins/aws/apigateway/apigatewayWafEnabled.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'API Gateway WAF Enabled',
+    category: 'API Gateway',
+    description: 'Ensures that API Gateway APIs are associated with a Web Application Firewall.',
+    more_info: 'API Gateway APIs should be associated with Web Application Firewall to ensure API security.',
```
Comment (dipsubha06) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:8]:
```suggestion
    more_info: 'API Gateway APIs should be associated with a Web Application Firewall to ensure API security.',
```

File: plugins/aws/apigateway/apigatewayWafEnabled.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+                if (!getStages.data.item || !getStages.data.item.length) {
+                    helpers.addResult(results, 0,
+                        'No API Gateway Rest API Stages found',
+                        region, resource);
+                    return cb();
+                }
+
+                getStages.data.item.forEach(stage => {
+                    if (!stage.webAclArn && stage.stageName) {
```
Comment (giorod3) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:61]: what happens here if there is a stage but it has no name?

File: plugins/aws/apigateway/apigatewayWafEnabled.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                        unassociatedStages.push(stage.stageName);
```
Comment (giorod3) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:62]:
```suggestion
                        unassociatedStages.push(stage.stageName ? stage.stageName : '');
```

File: plugins/aws/apigateway/apigatewayWafEnabled.js (hunk 3 above)
Comment (giorod3) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:61]:
```suggestion
                    if (!stage.webAclArn) {
```

File: plugins/aws/apigateway/apigatewayWafEnabled.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+                if (!unassociatedStages.length) {
+                    helpers.addResult(results, 0,
+                        `API Gateway API "${api.id}" has WAF enabled for all stages`,
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        `API Gateway API "${api.id}" does not have WAF enabled for these stages: ${unassociatedStages.join(', ')}`,
+                        region, resource);
+                }
```
Comment (giorod3) [inline plugins/aws/apigateway/apigatewayWafEnabled.js:74]: make this per stage and remove the array

**Review by dipsubha06 — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**